### PR TITLE
[AIRFLOW-1723] Support sendgrid in email backend

### PR DESCRIPTION
**Pull request by fenglu-db**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1723

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
  Extend airflow email backend to support sending emails via sendgrid. 

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
  tests.utils.test_email to ensure the mail data is constructed correctly.

### CODE REVIEW

**Comment by andyxhadji** [comment]
May want to include sendgrid package install in setup.py and requirements.txt (for ci)

**Comment by fenglu-db** [comment]
@andyxhadji good point, was about to add these :+1:

File: airflow/utils/email.py [hunk 1]
```diff
@@ -21,13 +21,16 @@
...
+import mimetypes
 import os
+import sendgrid
 import smtplib
 
 from email.mime.text import MIMEText
 from email.mime.multipart import MIMEMultipart
 from email.mime.application import MIMEApplication
 from email.utils import formatdate
+from sendgrid.helpers.mail import *
```
Comment (ashb) [inline airflow/utils/email.py]: Importing * is considered bad form in python -- it makes it hard to know what comes from which module.

Comment (fenglu-db) [reply airflow/utils/email.py]: Good catch, updated.

**Comment by fenglu-db** [comment]
@criccomini mind taking a quick look?

**Comment by msumit** [comment]
Started a mail chain about reverting this commit from the master. 

cc - @criccomini @Fokko @mistercrunch @r39132 and @bolkedebruin

**Comment by criccomini** [comment]
I'm fine with reverting and doing a plugin instead, as discussed on email.

**Pull request closed without merging**